### ISSUE TITLE
Login mechanism for custom data-sets

### ISSUE THREAD

**Issue by sumit3203** [zibneuro/fishexplorer#22]
User often want to load their data-sets but don't want to share it with other users

**Comment by owenrandlett** [comment]
I just had another request from a user who was trying to use the "upload" feature on the old site (https://engertlab.fas.harvard.edu/Z-Brain/home/) . Do you have an idea for a timeline for when something similar will be possible?

**Comment by sumit3203** [comment]
Hi Owen, Functionality is already implemented and available.  
1. User need to signin/register by clicking signin button on top.
2. Once signin is done, the user will see load dataset button in as shown in pic
![loaddataset](https://user-images.githubusercontent.com/9287945/82448399-f479e080-9aa9-11ea-9de6-fab921be72d2.png)
. 
3. Click load dataset button , it will allow user to select dataset file. Once dataset is uploaded, list of datasets will be updated.
4. User can even delete its dataset by clicking cross button in front of dataset.

Things that still need to fix:
1. Progress bar to show loading of dataset.
2. Once user delete dataset, default dataset should be automatically selected.

Please test and comment

**Comment by owenrandlett** [comment]
OK great - thanks! What kind of files can be uploaded?

**Comment by sumit3203** [comment]
Currently, only tiff stack, with 621 width and 1406 with 138 slice can be uploaded. Do you think we should also allow to upload single slice or partial tiff stack like only 40 slices of 138?

**Comment by udhillon** [comment]
hello! I am the person who was asking about the upload feature. If it's an easy thing for you to add, it would be nice to have the option of uploading single slice and partial stacks.

**Comment by sumit3203** [comment]
Hi! I can add even that feature but it will take 1-2 days to finish it.

**Comment by owenrandlett** [comment]
If people upload a substack, how will you know where to put it within the
larger atlas volume?

**Comment by sumit3203** [comment]
User need to provide extra parameters i.e range of slice for partial stacks and slice number for single slice.

**Comment by owenrandlett** [comment]
right - that makes sense. I am just a bit worried that someone might try
and upload unregistered data. That doesnt really make any sense, so I think
there should be some kind of warning about how the data needs to be in the
correct coordinate space.

**Comment by sumit3203** [comment]
It totally make sense to give such a warning for unregistered data. I will include warning and even validations etc

**Comment by udhillon** [comment]
Hello,
I am attempting to upload an image that is 35 stacks and in the .tiff format. The website seems to accept it, but nothing changes on the viewer (just shows reference brain). Is there something I am doing wrong?

**Comment by sumit3203** [comment]
Hi, I checked your dataset. Yes, website accepted it but there are two issues with your dataset that's why slice is not appearing on the viewer.
1. You are trying to upload unregistered dataset. Only tiff stack, with 621 width and 1406 height with 138 slice can be uploaded. You must register your dataset on  z-brain reference t-erk ( (https://engertlab.fas.harvard.edu/Z-Brain/download; Reference brain).) using ANTS or CMTK. The basic tutorial for ANT registration is provided here (https://zebrafishatlas.zib.de/tutorial/image-registration.html). Feel free to ask questions related to ANT's registration.

2.  Secondly, you are trying to upload multi-channel stack. We are currently supporting single channel. I can extend it to multichannel. Please upload only single channel for time-being.

Sorry, I will update platform tomorrow with new updates then it will give error if user try to upload unregistered data or multichannel data.

**Comment by udhillon** [comment]
Thank you very much for your help!

**Comment by udhillon** [comment]
Sorry, I have another question... is it 621x1406 pixels or micrometres? Thank you!

**Comment by owenrandlett** [comment]
pixels

**Comment by udhillon** [comment]
Hello,
Sorry to bother you with so many questions, but I am attempting to register my images with the ANT, and I am wondering what software would be appropriate (I have a Windows computer). I have never used ANT before so please excuse my lack of knowledge.

**Comment by sumit3203** [comment]
Hi,
It's very difficult to install ANT's on windows computer. We are using ANT's because there are lot of registration option with ANT's and even results are better. For windows, there are two options: CMTK(https://www.nitrc.org/projects/cmtk/) and Elastix(https://elastix.lumc.nl/download.php). It is better to start with CMTK because window's binary is available.
